Add fetcher test info to developer documentation (#11873)
JabRef/jabref: docs/code-howtos/fetchers.md

```

When executing `./gradlew run`, gradle executes `processResources` and populates `build/build.properties` accordingly. However, when working directly in the IDE, Eclipse keeps reading `build.properties` from `src/main/resources`. In IntelliJ, the task `JabRef Main` is executing `./gradlew processResources` before running JabRef from the IDE to ensure the `build.properties` is properly populated.

## Committing and pushing changes to fetcher files

> Fetcher tests are run when pushing changes from any file in the `src/main/java/org/jabref/logic/importer/fetcher/` directory.

Reviewer: ```suggestion
Fetcher tests are run when a PR contains changes touching any file in the `src/main/java/org/jabref/logic/importer/fetcher/` directory.
```

> Since these tests rely on remote services, some tests may fail due to the network or the external server.

Reviewer: ```suggestion
Since these tests rely on remote services, some of them may fail due to the network or the external server.
```
